src/dsp_tools/analyse_xml_data/construct_and_analyze_graph.py

```diff
@@ -152,38 +153,40 @@ def _remove_leaf_nodes(
 def _find_cheapest_outgoing_links(
     graph: rx.PyDiGraph[Any, Any],
     cycle: list[tuple[int, int]],
-    edge_list: list[tuple[int, int, XMLLink | ResptrLink]],
-) -> list[tuple[int, int, XMLLink | ResptrLink]]:
+    edges: list[Edge],
+) -> list[Edge]:
     """
     This function searches for the nodes whose outgoing links should be removed in order to break the cycle.
     It calculates which links between the resources create the smallest stash.
 
     Args:
         graph: graph
         cycle: the list with (source, target) for each edge in the cycle
-        edge_list: list of all the edges that were in the original graph
+        edges: edges in the graph (contains info about source node, target node, and link info)
 
     Returns:
-        - A list with the links that should be stashed.
-        - It contains all the edges connecting the two nodes.
+        The edges (i.e. links) that should be stashed (containing all the edges connecting the two nodes)
     """
-    costs = []
+    Cost = namedtuple("Cost", ["source", "target", "node_value", "edges_out"])
```

Comment:
I'm not a fan of mixing named tuples and dataclasses. In this case, does this really add value, here?

Reply:
The added value comes some lines further down, where I can refer to the names of the `Cost` object, instead of indexing it:

Now:

```
... if x.source_rx_index == cheapest_cost.source and x.target_rx_index == cheapest_cost.target
```

Without namedtuple:

```
... if x.source_rx_index == cheapest_cost[0] and x.target_rx_index == cheapest_cost[1]
```

A dataclass would introduce too much overhead, but I wanted to have a one-liner

Reply:
I actually liked the old one better, there. I don't think we should go so far as to completely avoid index access - and with the named arguments, it actually doesn't fit on one line anymore.

Reply:
It's now a dataclass (defined outside of the function, which improves performance)

Reply:
Interesting... did not expect that to make it faster. But all the better